### ISSUE TITLE
Project Log (First stage)

### ISSUE THREAD

**Issue by swiri021** [BI-academic/nextflow-g#2]
- Main Dashboard Brainstorming
    - Pipeline group status (# of Running, # of Failed, # of Completed - Chart)
    - Latest pipeline status
    - Number of pipelines
    - Spec (?), notification for overflow pipeline (Optional)
    
- Functional list (Pipeline = Nextflow)
    - Login process
    - Pipeline installation process
    - Pipeline execution
    - Pipeline monitor, Pipeline log managing(optional)
       - Basic Status (RUNNING, COMPLETED, FAILED, UNKNOWN, PENDING) (Primary goal),
       - Background status checking (**Celery ?**, Redis + RabbiMQ)
       - Log streaming (Not realtime, Updating with regular interval(30s, 1m)) (Secondary goal)
    - Pipeline parameters management by profile, Profile config file loading (json, yaml)
    - report.html, time.html viewer
    - Nextflow config parsing (cpus, mem, hour) and checking with current local capacity

- Essential background
    - Nextflow (Snakemake)
    - SQL
    - Database
    - Sqlite3

**Comment by swiri021** [comment]
Engineering stack: VSCode, Conda, Docker, Nextflow, Flask, Sqlite, python3.9, Celery, Localstack

**Comment by duaghk** [comment]
Engineering stack: 
Currently: VSCode, Singularity, Django, SQLite, MariaDB, Snakemake, Python 3.9~3.12, 
Not currently: Conda, Docker, AWS batch, AWS step-functions

**Comment by lioseo** [comment]
Engineering stack: VSCode, Virtualenv, Docker, python3.8~10,  Localstack

**Comment by duaghk** [comment]
저는 Snakemake로만 Pipeline 구성을 해봐서 Nextflow 문법을 좀 익히기 위해서 Nextflow pipeline 구축 sub로 참여해보고 싶습니다. 
추가로 Background status check -> database 전송을 위해 이쪽을 메인으로 해보고 싶습니다.

**Comment by jeong2624** [comment]
Engineering stack: Jupyter, python 3.9, conda (mamba)

**Comment by swiri021** [comment]
개발 담당 
- @duaghk  : Nextflow pipeline (Bulk RNASeq example), Pipeline monitor
- @jeong2624 : Nextflow pipeline (DNASeq example), Main dashboard development
- @lioseo  : Pipeline installation(registration) process
- @swiri021 : Pipeline execution, Pipeline parameter management, project management

**Comment by lioseo** [comment]
Engineering stack: VSCode, Virtualenv, Docker, Sqlite, python3.8~3.10, Localstack
not currently: Conda, Flask, mongoDB, Go

**Comment by lioseo** [comment]
전체적으로 workflow 이해를 위함이고, 제가 이해한 부분인데 맞는지 피드백 부탁 드립니다. 그리고 무엇이 더 필요한지 등.. 빠진 부분 있다면 추가해주셔도 좋을 것 같습니다. 

#pipeline == nextflow manager 서비스 사용자 시나리오

1. 웹에 접속한다. 
2. nextflow standalone 인스톨한다.
(1회만 인스톨? 일반적인 인스톨개념이 아닌config 설정 등 standalone버전 설정이므로 처음 실행일지? toml파일은 사용자가 별도 저장한 파일에서 설정하고 로드)
3. nextflow standalone 실행한다. (처음실행, 파이프라인이 다 끝나는 부분이 1회 )
4. nextflow 실행상태를 보여준다. 
5. nextflow 완료한다. 파일을 지정된 위치에 저장한다.  (결과는 분석영역일듯하나 잘 나오는지는 검증 필요 )

**Comment by swiri021** [comment]
> 전체적으로 workflow 이해를 위함이고, 제가 이해한 부분인데 맞는지 피드백 부탁 드립니다. 그리고 무엇이 더 필요한지 등.. 빠진 부분 있다면 추가해주셔도 좋을 것 같습니다. 
> 
> 
> 
> #pipeline == nextflow manager 서비스 사용자 시나리오
> 
> 
> 
> 1. 웹에 접속한다. 
> 
> 2. nextflow standalone 인스톨한다.
> 
> (1회만 인스톨? 일반적인 인스톨개념이 아닌config 설정 등 standalone버전 설정이므로 처음 실행일지? toml파일은 사용자가 별도 저장한 파일에서 설정하고 로드)
> 
> 3. nextflow standalone 실행한다. (처음실행, 파이프라인이 다 끝나는 부분이 1회 )
> 
> 4. nextflow 실행상태를 보여준다. 
> 
> 5. nextflow 완료한다. 파일을 지정된 위치에 저장한다.  (결과는 분석영역일듯하나 잘 나오는지는 검증 필요 )

네, 맞습니다. 파이프라인 설치는 한번만 이뤄지고 추후엔 저장된 파이프라인으로 인풋파일만 바꿔서 실행되는 형태여야 합니다.

**Comment by swiri021** [comment]
### 회의록 12/6/2024
- Progress sharing
- workflow_parameters.json : https://docs.aws.amazon.com/omics/latest/dev/parameter-templates.html
- nextflow config: `nextflow config -a -o json`
- Nextflow instruction and questions

**Comment by swiri021** [comment]
### 회의록 1/3/2024
- Progress sharing
- Branching discussion (Feature branch to dev-v0.0.1)
- DNAseq RNAseq pipeline repositories
- Pipeline sub-applications composition
- rnaseq nf-core introduction: [link](https://github.com/nf-core/rnaseq/tree/master)
- Meeting schedule update(?) on Feb

**Comment by duaghk** [comment]
# 회의록 3/9/2025

- Test pipeline(RNAseq) - config docker change need
- Celery - Job status + NF status double check
- DNAseq pipeline fix 
- Proxy error -  fix

**Comment by swiri021** [comment]
### 회의록 4/4/2025
- Schedule delay issue : Targeting to the end of August 2025, 10AM, Saturday fixed(Korea)
-  Jun's computer issue and keep doing Celery test until the end of April
- Error solving(GVCF generation) for Nextflow pipeline

**Comment by swiri021** [comment]
### 회의록 4/26/2025
- **Schedule delay issue**
- DNA Pipeline development has been completed 
- Local saving for pipeline codes and updating DB tables

**Comment by swiri021** [comment]
### 회의록 5/17/2025
- 정준, 정운님 코드 설명 필요
- 혜영님 코드 설명
- workflow_parameter.json instruction

```json
  {
    "UserParameter1": {
       "description": "this parameter is required",
       "default": "100"
     },
    "UserParameter2": {
       "description": "this parameter is also required",
       "default": "true"
    }
  }   
```
